### PR TITLE
NMS-15423: Add database report example

### PR DESCRIPTION
**Pull request by emarsh1995**
- Add `operation/database-reports/report-example.png`
- Update `operation/deep-dive/database-reports/database.adoc`, `/template-introduction.adoc`

### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the Jira link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the Jira issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here, or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready. Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15423

### CODE REVIEW

File: docs/modules/operation/pages/deep-dive/database-reports/database.adoc [hunk 1]
```diff
@@ -317,8 +315,9 @@ For more details, see <<ga-database-reports-interface-descriptor, Interface desc
 [[ga-database-reports-node-source-descriptor]]
 === Node source descriptors
 
-A node is addressed by a node source descriptor, which references the node using either the foreign source and foreign ID, or the node ID.
-If `storeByForeignSource` is enabled, you can address the node only using its foreign source and foreign ID (see https://opennms.discourse.group/t/storing-data-with-foreign-sources/2057[Storing data with foreign sources] on Discourse).
+A node is addressed by a node source descriptor.
+The node source descriptor references a node using either its foreign source and foreign ID, or it node ID.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/database-reports/database.adoc:319]:
```suggestion
The node source descriptor references a node using either its foreign source and foreign ID, or its node ID.
```

**Review by Bonrob2 — COMMENTED** [review]
Some minor text changes, and a comment to display a sample report for each report template.

**Comment by emarsh1995** [comment]
I'll be updating this locally to include more examples of reports, then I'll push an update.

**Review by Bonrob2 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**